### ISSUE TITLE
Clarification on Standard Training Configuration vs. Best Ablation Results

### ISSUE THREAD

**Issue by Xiaoqiong-Liu** [valeoai/NAF#7]
Thank you for the impressive work on NAF. I am currently reviewing the results for a comparative study and noticed a point of clarification regarding the training configuration.

In Table 10, the results indicate that training the Dual-Branch Encoder with DINOv2-R-B features yields the highest average mIoU (61.72) across test backbones, surpassing the DINOv3-B configuration (60.41). However, the paper suggests that DINOv3-B is the "standard training configuration" used for the main results in Table 2.

Could you please confirm:

Were the results in Table 2 indeed produced using the model trained specifically on DINOv3-B features?

If the DINOv2-R-B training setup yields better zero-shot generalization, is there a version of Table 2 (covering VOC and NYUv2) that reflects these "best-case" ablation performance numbers?

Knowing the absolute performance ceiling of NAF would be very helpful for my research. Thank you for your time!

X.

**Comment by LoickCh** [comment]
Hi,

Thank you for your interest in our work. To clarify:
- We indeed select DINOv3-B as the standard backbone. So in all tables except 'backbone ablation table' our unique model was trained using DINOv3-B.

About the justification of the choice of the backbone, we observe that during our stage 2, since DINOv3 is somewhat scale-invariant, we obtained better results than performing stage 2 with DINOv2-R as you can observe in the following table! In order to have fair comparisons, and to verify our gains was not due to the choice of the backbone we have trained AnyUp and JAFAR in the same training protocol and evaluated both on Cityscapes/VOC. Here is the table:

<img width="893" height="387" alt="Image" src="https://github.com/user-attachments/assets/7fd06555-e25e-41d9-acf0-b4d26ac5698b" />

Note that: _The values shown in blue (resp. purple) are consistent with those of Tab. 3 (resp. Supp. Tab. 2). These comparisons demonstrate that the performance gains are not driven by the choice of training VFM, refinement stage, or resolution. Most notably, we observe that NAF trained for only 25k steps without any refinement stage outperforms the official AnyUp model trained for 100k steps with its refinement stage (using the same data, DINOv2-S training VFM, and resolution)_

Do you have any remaining questions?

**Comment by Xiaoqiong-Liu** [comment]
Thank you for the clarification.  Now I have a clearer picture of it.  Below is my test result with NAF trained with DINOv2-R-B and AnyUp multi-backbone official release( I believe there's no stage 2 in current version of training, I will double-check)

Model | Backbone | Accuracy | IoU
-- | -- | -- | --
anyup_multi_backbone | dinov2_small | 0.9619 | 0.8411
anyup_multi_backbone | dinov2_r | 0.9661 | 0.8553
anyup_multi_backbone | franca | 0.9605 | 0.8323
naf_ckpt_dinov2_r | dinov2_small | 0.9637 | 0.8475
anyup_multi_backbone | dinov3_base | 0.9672 | 0.8614
naf_ckpt_dinov2_r | dinov3_base | 0.9718 | 0.8770
anyup_multi_backbone | radiov2.5-B | 0.9652 | 0.8567
naf_ckpt_dinov2_r | franca | 0.9639 | 0.8433
naf_ckpt_dinov2_r | radiov2.5-B | 0.9682 | 0.8670

So I believe trained with DINOv3-B perform better on itself.  And your above result seems to be align with it.

Another request: Is it possible to share the depth probing code and configuration?

**Comment by LoickCh** [comment]
Hi,
So if I read correctly when you compare: AnyUp / NAF, you have:

Dinov2-S: 84.11 / 84.75 (+0.64)
Franca: 83.23 / 84.33 (+0.10)
Dinov3: 86.14 / 87.70 (+1.56)
Radio: 85.67 / 86.70 (+1.03)

That's confirm our results about NAF leading to better scores. That's cool. 
Have you also compared the official checkpoints? Because in our official checkpoint we have a second training stage (as mentioned in the previous table) and AnyUp may also have one.

I will look for the depth probing code and I keep you in touch, but in the spirit, it is very close to what was done for JAFAR.

**Comment by LoickCh** [comment]
Btw, during how many steps have you trained the models? AnyUp requires x4 more steps during training, I am just curious about your training protocol

**Comment by Xiaoqiong-Liu** [comment]
Yes, I have compared with your offical NAF checkpoints(which may have a second training stage).  I am not sure why my franca-B test result is much higher than reported.  See:


Model | Backbone | Dataset | Accuracy | IoU
-- | -- | -- | -- | --
naf | dinov2_small | voc | 0.9633 | 0.8466
naf | dinov2_r | voc | 0.9682 | 0.8628
naf | franca | voc | 0.9641 | 0.8440
naf | dinov3_base | voc | 0.9720 | 0.8782
naf | radiov2.5-B | voc | 0.9681 | 0.8663

For the naf_ckpt_dinov2_r model I mentioned above, I am training 25k without a second stage. The resolution should be close to 512(but I have a rounding mechanism for multi-backbone purpose).

Thanks for the clarification of depth probing protocal, I follow Jafar but have much lower test result on all DINOv3-B probing (including bilinear). So I am looking for a reference.

**Comment by LoickCh** [comment]
For Franca, the authors have updated their weights, I was evaluating the old ones, and you are probably evaluating the newer ones. 
I will try to look for depth probing as soon as possible.

**Comment by LoickCh** [comment]
Hi,

Sorry for the late reply. I was quite busy. I have added a file called eval_depth_probing.py. I think this is the one I used for evaluating depth, but since it has been a long time, do not hesitate to tell me if you have any error.
Best,
Loïck
